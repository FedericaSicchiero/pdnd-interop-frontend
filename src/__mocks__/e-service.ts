import { EServiceReadType } from '../../types'

export const eserviceDraft: EServiceReadType = {
  producer: {
    id: 'djofsi-sdfjdsi-djsfs',
    name: 'Comune di Bologna',
  },
  name: 'Nome e-service',
  description: 'Descrizione e-service',
  technology: 'REST',
  id: 'sdjof-sdfjdspof-dsfdsjf',
  state: 'DRAFT',
  descriptors: [],
  attributes: {
    verified: [],
    declared: [],
    certified: [
      {
        single: {
          id: 'dsdsld-dsdlds-lsdasdas',
          creationTime: '2022-02-28T16:16:18.879093Z',
          explicitAttributeVerification: false,
          verified: false,
          origin: 'dfkdsfk',
          code: 'dfjdso',
          name: 'Attributo 1',
          description: 'Descrizione attributo 1',
        },
      },
    ],
  },
}
