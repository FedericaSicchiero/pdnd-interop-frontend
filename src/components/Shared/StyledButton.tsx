import React, { FunctionComponent } from 'react'
import { Button } from '@mui/material'

export const StyledButton: FunctionComponent<any> = React.forwardRef(
  ({ children, ...props }, ref) => {
    return (
      <Button {...props} ref={ref}>
        {children}
      </Button>
    )
  }
)
