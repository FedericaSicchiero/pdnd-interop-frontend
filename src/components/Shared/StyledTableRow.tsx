import React, { FunctionComponent, ReactElement } from 'react'
import { TableCell, TableRow, Typography } from '@mui/material'

type LabelCell = {
  label: string | JSX.Element
  tooltip?: ReactElement
}

type CustomCell = {
  custom: React.ReactNode
}

type Cell = LabelCell | CustomCell

type StyledTableRowProps = {
  cellData: Array<Cell>
}

export const StyledTableRow: FunctionComponent<StyledTableRowProps> = ({ cellData, children }) => {
  return (
    <TableRow sx={{ bgcolor: 'common.white' }}>
      {cellData.map((cell, i) => {
        return (
          <TableCell key={i} sx={{ py: 2 }}>
            {'custom' in cell ? (
              cell.custom
            ) : (
              <Typography component="span" sx={{ display: 'inline-block' }} variant="body2">
                {cell.label}
                {cell.tooltip}
              </Typography>
            )}
          </TableCell>
        )
      })}
      {children && (
        <TableCell sx={{ minWidth: '15rem' }} align="right">
          {children}
        </TableCell>
      )}
    </TableRow>
  )
}
