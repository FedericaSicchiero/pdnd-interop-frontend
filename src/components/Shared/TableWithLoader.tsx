import React, { Children, FunctionComponent } from 'react'
import {
  TableContainer,
  Table,
  TableBody,
  TableHead,
  TableRow,
  TableCell,
  Grid,
  Alert,
} from '@mui/material'
import { Box } from '@mui/system'
import { LoadingWithMessage } from './LoadingWithMessage'

type TableWithLoaderProps = {
  loadingText: string | null
  noDataLabel?: string
  error?: Error
  headData: Array<string>
  viewType?: 'table' | 'grid'
}

export const TableWithLoader: FunctionComponent<TableWithLoaderProps> = ({
  loadingText,
  noDataLabel = 'Questa ricerca non ha prodotto risultati',
  error,
  headData,
  viewType = 'table',
  children,
}) => {
  if (error) {
    return (
      <Box sx={{ my: 4, p: 2 }} bgcolor="error.main" color="common.white">
        C’è stato un errore, e non è stato possibile reperire le informazioni richieste. Per favore,
        riprova più tardi
      </Box>
    )
  }

  if (loadingText) {
    return <LoadingWithMessage label={loadingText} transparentBackground={true} />
  }

  return viewType === 'table' ? (
    <TableContainer>
      <Table>
        <TableHead sx={{ bgcolor: 'background.default' }}>
          <TableRow>
            {headData.map((item, i) => (
              <TableCell key={i}>{item}</TableCell>
            ))}
          </TableRow>
        </TableHead>
        <TableBody sx={{ bgcolor: 'background.paper' }}>
          {children && Children.count(children) > 0 ? (
            children
          ) : (
            <TableRow>
              <TableCell colSpan={headData.length}>
                <Alert severity="info">{noDataLabel}</Alert>
              </TableCell>
            </TableRow>
          )}
        </TableBody>
      </Table>
    </TableContainer>
  ) : (
    <Grid container columnSpacing={2} rowSpacing={2} alignItems="stretch">
      {children && Children.count(children) > 0 ? (
        Children.map(children, (c, i) => (
          <Grid item xs={4} key={i}>
            {c}
          </Grid>
        ))
      ) : (
        <Grid item xs={12}>
          <Alert severity="info">{noDataLabel}</Alert>
        </Grid>
      )}
    </Grid>
  )
}
