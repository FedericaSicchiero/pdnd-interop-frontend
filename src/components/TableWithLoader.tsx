import React, { FunctionComponent } from 'react'
import { Table } from 'react-bootstrap'
import { LoadingWithMessage } from './LoadingWithMessage'
import { TablePagination } from './TablePagination'

type TableWithLoaderProps = {
  isLoading: boolean
  loadingLabel?: string
  headData: string[]
  pagination?: boolean
}

export const TableWithLoader: FunctionComponent<TableWithLoaderProps> = ({
  isLoading,
  loadingLabel,
  headData,
  children,
  pagination = false,
}) => {
  return isLoading ? (
    <LoadingWithMessage label={loadingLabel} />
  ) : (
    <div>
      {pagination && <TablePagination />}
      <Table striped>
        <thead className="bg-light text-dark">
          <tr>
            {headData.map((item, i) => (
              <th key={i}>{item.toUpperCase()}</th>
            ))}
          </tr>
        </thead>
        <tbody>{children}</tbody>
      </Table>
    </div>
  )
}
