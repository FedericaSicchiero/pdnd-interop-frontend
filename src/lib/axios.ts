import axios from 'axios'
import qs from 'qs'
import { v4 as uuidv4 } from 'uuid'
import { STORAGE_KEY_SESSION_TOKEN } from '@/config/constants'
import { storageRead } from '@/utils/storage.utils'
import { NotAuthorizedError, NotFoundError, ServerError } from '@/utils/errors.utils'

const axiosInstance = axios.create({
  paramsSerializer: {
    serialize: (params) => qs.stringify(params, { arrayFormat: 'comma' }),
  },
})

axiosInstance.interceptors.request.use(
  (config) => {
    const _config = { ...config, headers: { ...config.headers } }

    const sessionStorageToken = storageRead(STORAGE_KEY_SESSION_TOKEN, 'string')
    if (sessionStorageToken) {
      _config.headers.Authorization = `Bearer ${sessionStorageToken}`
    }

    _config.headers['X-Correlation-Id'] = uuidv4()

    return _config
  },
  (error) => {
    return Promise.reject(error)
  }
)

axiosInstance.interceptors.response.use(
  (response) => {
    return response
  },
  (error) => {
    const isAxiosError = axios.isAxiosError(error)
    if (isAxiosError && error.response?.status === 404) {
      return Promise.reject(new NotFoundError())
    }
    if (isAxiosError && error.response?.status === 401) {
      return Promise.reject(new NotAuthorizedError())
    }
    if (
      isAxiosError &&
      error.response?.status &&
      (error.response?.status >= 500 || error.response?.status === 400)
    ) {
      return Promise.reject(
        new ServerError(error.message, error.code, error.config, error.request, error.response)
      )
    }

    return Promise.reject(error)
  }
)

export default axiosInstance
