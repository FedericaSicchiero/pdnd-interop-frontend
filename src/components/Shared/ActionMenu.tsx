import React, { FunctionComponent, useContext, useRef, useState } from 'react'
import { MoreVert as MoreVertIcon } from '@mui/icons-material'
import { StyledButton } from './StyledButton'
import { Menu, MenuItem } from '@mui/material'
import { ActionProps, ExtendedMUIColor } from '../../../types'
import { TableActionMenuContext } from '../../lib/context'
import { Box } from '@mui/system'

type ActionMenuProps = {
  // The list of actions to display in the menu
  actions: Array<ActionProps>
  onOpen: (id: string) => void
  onClose: VoidFunction
  // The id of the menu currently clicked. In a table there may be many of them,
  // but only one is open at a time (or none, if the value is null)
  openMenuId: string | null
  // Only used for snapshot tests, to have a stable id
  snapshotTestInternalId?: string
  iconColor?: ExtendedMUIColor
}

export const ActionMenu: FunctionComponent<
  Omit<ActionMenuProps, 'onOpen' | 'onClose' | 'openMenuId'>
> = (props) => {
  const { tableActionMenu, setTableActionMenu } = useContext(TableActionMenuContext)

  const onOpen = (id: string) => {
    setTableActionMenu(id)
  }

  const onClose = () => {
    setTableActionMenu(null)
  }

  return (
    <ActionMenuComponent
      {...props}
      onOpen={onOpen}
      onClose={onClose}
      openMenuId={tableActionMenu}
    />
  )
}

// From https://stackoverflow.com/a/1349426
function uniqueString(length: number) {
  let result = ''
  const chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
  for (let i = 0; i < length; i++) {
    result += chars.charAt(Math.floor(Math.random() * chars.length))
  }
  return result
}

const ActionMenuComponent: FunctionComponent<ActionMenuProps> = ({
  actions,
  onOpen,
  onClose,
  openMenuId,
  snapshotTestInternalId,
  iconColor = 'primary',
}) => {
  // Needs to be state to avoid it changing on rerender
  const [id] = useState(snapshotTestInternalId || uniqueString(8))
  const anchorRef = useRef() as React.MutableRefObject<HTMLSpanElement>
  const anchorId = `basic-button-${id}`
  const menuId = `basic-menu-${id}`
  const open = Boolean(openMenuId !== null && openMenuId === anchorId)

  const handleClick = (event: React.SyntheticEvent) => {
    event.stopPropagation()
    event.preventDefault()
    onOpen(event.currentTarget.id)
  }

  if (!Boolean(actions.length > 0)) {
    // Used to keep the buttons visually aligned in case there is no ActionMenu
    return <Box component="span" sx={{ width: 70, display: 'inline-block' }} />
  }

  return (
    <React.Fragment>
      <StyledButton
        id={anchorId}
        aria-controls={menuId}
        aria-haspopup="true"
        aria-expanded={open ? 'true' : undefined}
        onClick={handleClick}
        color={iconColor}
      >
        <span ref={anchorRef}>
          <MoreVertIcon color="inherit" fontSize="small" />
        </span>
      </StyledButton>

      <Menu
        id={menuId}
        anchorEl={anchorRef.current}
        open={open}
        onClose={onClose}
        MenuListProps={{
          'aria-labelledby': 'basic-button',
        }}
        anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
        transformOrigin={{ vertical: 'top', horizontal: 'right' }}
      >
        {actions.map(({ onClick, label }, i) => {
          return (
            <MenuItem key={i} onClick={onClick}>
              {label}
            </MenuItem>
          )
        })}
      </Menu>
    </React.Fragment>
  )
}
