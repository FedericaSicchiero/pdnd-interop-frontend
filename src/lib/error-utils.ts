import { AxiosError, AxiosResponse } from 'axios'

export function isFetchError(resp: AxiosResponse | AxiosError) {
  // This typing sucks, improve it
  return (resp as AxiosError).isAxiosError
}

export function getFetchOutcome(resp: AxiosResponse | AxiosError) {
  return isFetchError(resp) ? 'error' : 'success'
}

export function axiosErrorToError(axiosError: AxiosError | undefined): Error | undefined {
  return axiosError && new Error(axiosError.message)
}
