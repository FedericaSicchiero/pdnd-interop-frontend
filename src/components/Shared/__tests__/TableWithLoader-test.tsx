import React from 'react'
import renderer from 'react-test-renderer'
import { TableWithLoader } from '../TableWithLoader'
import { AxiosError } from 'axios'
import { StyledTableRow } from '../StyledTableRow'
import noop from 'lodash/noop'
import { createMemoryHistory } from 'history'
import { AllTheProviders } from '../../../__mocks__/providers'
import { ActionMenu } from '../ActionMenu'
import { StyledButton } from '../StyledButton'
import { axiosErrorToError } from '../../../lib/error-utils'

type ExampleDatum = {
  name: string
  surname: string
  id: string
}
const headData = ['Nome', 'Cognome', '']
const rawData: Array<ExampleDatum> = [
  { name: 'Mario', surname: 'Rossi', id: 'rejsedf3-re4k-rew2-eoer' },
  { name: 'Teresa', surname: 'Verdi', id: 'dosf0i23-jkds-32jd-23rj' },
]

describe('Snapshot', () => {
  it('matches error', () => {
    const error: AxiosError = {
      isAxiosError: true,
      config: {},
      name: '',
      message: '',
      toJSON: () => ({}),
    }
    const component = renderer.create(
      <TableWithLoader loadingText={null} headData={headData} error={axiosErrorToError(error)} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches loading', () => {
    const component = renderer.create(
      <TableWithLoader loadingText="Sto caricando..." headData={headData} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches table without actions', () => {
    const getTableRow = (item: ExampleDatum, i: number) => (
      <StyledTableRow key={i} cellData={[{ label: item.name }, { label: item.surname }]} />
    )
    const component = renderer.create(
      <TableWithLoader loadingText={null} headData={headData}>
        {rawData.map((item, i) => getTableRow(item, i))}
      </TableWithLoader>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches table with single action button', () => {
    const history = createMemoryHistory()

    const getTableRow = (item: ExampleDatum, i: number) => (
      <StyledTableRow key={i} cellData={[{ label: item.name }, { label: item.surname }]}>
        <StyledButton
          onClick={() => {
            history.push(`/user/${item.id}`)
          }}
        >
          Ispeziona
        </StyledButton>
      </StyledTableRow>
    )
    const component = renderer.create(
      <AllTheProviders defaultHistory={history}>
        <TableWithLoader loadingText={null} headData={headData}>
          {rawData.map((item, i) => getTableRow(item, i))}
        </TableWithLoader>
      </AllTheProviders>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches table with single action button and action list button', () => {
    const history = createMemoryHistory()

    const getTableRow = (item: ExampleDatum, i: number) => (
      <StyledTableRow key={i} cellData={[{ label: item.name }, { label: item.surname }]}>
        <StyledButton
          onClick={() => {
            history.push(`/user/${item.id}`)
          }}
        >
          Ispeziona
        </StyledButton>
        <ActionMenu
          actions={[
            { onClick: noop, label: 'Azione 1' },
            { onClick: noop, label: 'Azione 2' },
          ]}
          snapshotTestInternalId="1"
        />
      </StyledTableRow>
    )
    const component = renderer.create(
      <AllTheProviders defaultHistory={history}>
        <TableWithLoader loadingText={null} headData={headData}>
          {rawData.map((item, i) => getTableRow(item, i))}
        </TableWithLoader>
      </AllTheProviders>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
