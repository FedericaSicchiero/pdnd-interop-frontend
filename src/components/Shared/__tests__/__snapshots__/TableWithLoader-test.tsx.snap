// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot matches error 1`] = `
<div
  className="MuiBox-root css-1qm4nww"
>
  C’è stato un errore, e non è stato possibile reperire le informazioni richieste. Per favore, riprova più tardi
</div>
`;

exports[`Snapshot matches loading 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w3q8k3-MuiPaper-root"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
  <p
    className="MuiTypography-root MuiTypography-body2 css-tl9904-MuiTypography-root"
  >
    Sto caricando...
  </p>
</div>
`;

exports[`Snapshot matches table with single action button 1`] = `
<div
  className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
>
  <table
    className="MuiTable-root css-rqglhn-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root css-whv87p-MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Nome
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Cognome
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-1nq9kkf-MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root css-1p72hwd-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1rpd3i8-MuiTableCell-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-1h9j9ew-MuiTypography-root"
          >
            Mario
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1rpd3i8-MuiTableCell-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-1h9j9ew-MuiTypography-root"
          >
            Rossi
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-2u9kxv-MuiTableCell-root"
        >
          <button
            className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Ispeziona
          </button>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root css-1p72hwd-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1rpd3i8-MuiTableCell-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-1h9j9ew-MuiTypography-root"
          >
            Teresa
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1rpd3i8-MuiTableCell-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-1h9j9ew-MuiTypography-root"
          >
            Verdi
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-2u9kxv-MuiTableCell-root"
        >
          <button
            className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Ispeziona
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Snapshot matches table with single action button and action list button 1`] = `
<div
  className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
>
  <table
    className="MuiTable-root css-rqglhn-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root css-whv87p-MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Nome
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Cognome
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-1nq9kkf-MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root css-1p72hwd-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1rpd3i8-MuiTableCell-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-1h9j9ew-MuiTypography-root"
          >
            Mario
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1rpd3i8-MuiTableCell-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-1h9j9ew-MuiTypography-root"
          >
            Rossi
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-2u9kxv-MuiTableCell-root"
        >
          <button
            className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Ispeziona
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="basic-menu-1"
            aria-haspopup="true"
            className="MuiButton-root MuiButton-naked MuiButton-nakedPrimary MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButtonBase-root  css-19lhi6j-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            id="basic-button-1"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                data-testid="MoreVertIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
          </button>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root css-1p72hwd-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1rpd3i8-MuiTableCell-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-1h9j9ew-MuiTypography-root"
          >
            Teresa
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1rpd3i8-MuiTableCell-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-1h9j9ew-MuiTypography-root"
          >
            Verdi
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-2u9kxv-MuiTableCell-root"
        >
          <button
            className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Ispeziona
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="basic-menu-1"
            aria-haspopup="true"
            className="MuiButton-root MuiButton-naked MuiButton-nakedPrimary MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButtonBase-root  css-19lhi6j-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            id="basic-button-1"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                data-testid="MoreVertIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Snapshot matches table without actions 1`] = `
<div
  className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
>
  <table
    className="MuiTable-root css-rqglhn-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root css-whv87p-MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Nome
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Cognome
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-1nq9kkf-MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root css-1p72hwd-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1rpd3i8-MuiTableCell-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-1h9j9ew-MuiTypography-root"
          >
            Mario
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1rpd3i8-MuiTableCell-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-1h9j9ew-MuiTypography-root"
          >
            Rossi
          </span>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root css-1p72hwd-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1rpd3i8-MuiTableCell-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-1h9j9ew-MuiTypography-root"
          >
            Teresa
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1rpd3i8-MuiTableCell-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-1h9j9ew-MuiTypography-root"
          >
            Verdi
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
